<?xml version="1.0"?>

<launch>

  <arg name="standalone_mode" default="false" description="launches only the planner if true"/>

  <node pkg="mpc_planner_autoware" exec="autoware_planner" name="autoware_planner" respawn="false" output="screen">
    <param from="$(find-pkg-share mpc_planner_autoware)/config/ros2_guidance_planner.yaml"/>
    <param from="$(find-pkg-share mpc_planner_autoware)/config/autotune_guidance.yaml"/>
    <remap from="~/input/state" to="/localization/kinematic_state"/>
    <remap from="~/input/steering" to="/vehicle/status/steering_status"/>
    <remap from="~/input/goal" to="/goal_pose"/>
    <remap from="~/input/reference_path" to="roadmap/path_with_lane_id"/>
    <remap from="~/input/simulated_obstacles" to="/pedestrian_simulator/trajectory_predictions"/>
    <remap from="~/input/obstacles" to="/perception/object_recognition/tracking/objects"/>
    <remap from="~/input/autoware_status" to="/api/operation_mode/state"/>
    <remap from="~/output/trajectory" to="/planning/scenario_planning/motion_velocity_smoother/trajectory"/>
  </node>

  <group unless="$(var standalone_mode)">
    <include file="$(find-pkg-share mpc_planner_autoware)/launch/path_and_pedestrians.launch"/>
  </group>

</launch>  
